### PR TITLE
mark all tests that interact with external services/docker as expensive

### PR DESCRIPTION
**Pull request by k-anshul**
INSERT DESCRIPTION HERE

**Checklist:**
- [ ] Covered by tests
- [ ] Ran it and it works as intended
- [ ] Reviewed the diff before requesting a review
- [ ] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [ ] I'm proud of this work!

### CODE REVIEW

**Comment by pjain1** [comment]
Do we run expensive tests in PR CI?

**Comment by k-anshul** [comment]
> Do we run expensive tests in PR CI?

Yeah Kasper set a workflow to run all expensive tests daily.

File: admin/pkg/pgtestcontainer/pgtestcontainer.go [hunk 1]
```diff
@@ -18,6 +19,7 @@ type Container struct {
 
 // New creates and starts a new Postgres testcontainer. Call Terminate() when done using it.
 func New(t *testing.T) Container {
+	testmode.Expensive(t)
```
Comment (begelundmuller) [inline admin/pkg/pgtestcontainer/pgtestcontainer.go:22]: I would like to keep the Postgres tests on the `admin` service. Almost every test in the `admin` service needs Postgres, so if it's disabled, we will basically not be running any `admin` service tests in normal CI. Postgres is pretty cheap to run even in tests, so I think it's okay to have it.

File: runtime/drivers/clickhouse/testclickhouse/testclickhouse.go [hunk 2]
```diff
@@ -27,6 +29,8 @@ type TestingT interface {
 // It returns the DSN for connecting to the container.
 // The container is automatically terminated when the test ends.
 func Start(t TestingT) string {
+	testmode.Expensive(t)
```
Comment (begelundmuller) [inline runtime/drivers/clickhouse/testclickhouse/testclickhouse.go:32]: I agree on marking Clickhouse cluster tests, but wondering if we should keep single-node Clickhouse tests? Since Clickhouse support is also essential for us and we should avoid breaking it.

File: runtime/resolvers/testdata/metrics_geospatial.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+expensive: true
```
Comment (begelundmuller) [inline runtime/resolvers/testdata/metrics_geospatial.yaml:1]: This is basically disabling all metrics tests in CI. I think we can disable for rare connectors, but I think we should keep running for common OLAPs.

**Comment by pjain1** [comment]
I would like the resolvers tests to run on each PR, whats the issue if its taking 5-10 mins more? I think they are very critical to catch issues related to dashboard queries. I don't think its good to retrospectively wait for it to run daily then raise follow up PRs.

**Comment by k-anshul** [comment]
> I would like the resolvers tests to run on each PR, whats the issue if its taking 5-10 mins more? I think they are very critical to catch issues related to dashboard queries. I don't think its good to retrospectively wait for it to run daily then raise follow up PRs.

But wouldn't you want to run tests locally for packages where you are making changes. The CI/CD loop is very time consuming.

**Comment by k-anshul** [comment]
> I'm a little uncomfortable with things like disabling Postgres in the admin service or single-node Clickhouse in the runtime. Because those are essential features for those services, and would require us to do manual runs on a large fraction of PRs (which people would forget).
> 
> If the performance penalty of running Postgres/Clickhouse tests is that big, I think we should instead consider using path rules for these common cases (e.g. if you only change code in `cli`, we don't need to run `runtime` or `admin` tests).

I often find it frustrating to wait for upto 20 minutes for CI/CD tests to finish. The motivation here was to see how much we can save by removing all dependencies. 
It seems like we aren't saving much. It took 16m here as compared to average of 18m for go tests in last week. 

Will investigate other approaches like path rules, concurrent tests etc.

**Comment by pjain1** [comment]
> But wouldn't you want to run tests locally for packages where you are making changes. The CI/CD loop is very time consuming.

I don't run full test suite everytime locally, find it convenient to use CI for this if making small changes and pushing, also  have to unmark the test as expensive which can be forgotten. Sometimes we may make a small change and assume its an unrelated change but may affect correctness. Probably a better approach can be taken here if we are not saving much.

**Comment by k-anshul** [comment]
But you don't need to run full test suite only the ones where you are making changes. I believe that is much faster loop than waiting for CI/CD to finish.
Given our code is not strongly coupled, I would expect that to be sufficient in majority, if not all, cases.

> also have to unmark the test as expensive which can be forgotten

You don't need to make code changes. Can set `RILL_RUNTIME_TEST_MODE=expensive` env. For eg: `RILL_RUNTIME_TEST_MODE=expensive go test -run github.com/rilldata/rill/runtime/pkg/rduckdb -v` 

I strongly believe that there is a lot of merit in making our CI/CD loop faster though we can decide on approaches that suits everybody and helps us to discover bugs faster.

**Comment by k-anshul** [comment]
Closing this PR for now and investigate alternate approaches.
